src/bin/list_store.rs

```diff
@@ -0,0 +1,311 @@
+        },
+        Data {
+            fixed: true,
+            number: 56355,
+            severity: "Normal".to_string(),
+            description: "GtkLabel - Not all changes propagate correctly".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 50055,
+            severity: "Normal".to_string(),
+            description: "Rework width/height computations for TreeView".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 58278,
+            severity: "Normal".to_string(),
+            description: "gtk_dialog_set_response_sensitive () doesn't work".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 55767,
+            severity: "Normal".to_string(),
+            description: "Getters for all setters".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 56925,
+            severity: "Normal".to_string(),
+            description: "Gtkcalender size".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 56221,
+            severity: "Normal".to_string(),
+            description: "Selectable label needs right-click copy menu".to_string(),
+        },
+        Data {
+            fixed: true,
+            number: 50939,
+            severity: "Normal".to_string(),
+            description: "Add shift clicking to GtkTextView".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 6112,
+            severity: "Normal".to_string(),
+            description: "netscape-like collapsable toolbars".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 1,
+            severity: "Normal".to_string(),
+            description: "First bug :=)".to_string(),
+        },
+    ];
+
+    let store = gtk::ListStore::new(&col_types);
+
+    let col_indices: [u32; 8] = [0, 1, 2, 3, 4, 5, 6, 7];
+
+    for (d_idx, d) in data.iter().enumerate() {
+        let icon_name = if d_idx == 1 || d_idx == 3 {
+            "battery-caution-charging-symbolic"
+        } else {
+            ""
+        };
+
+        let sensitive = d_idx != 3;
+
+        let values: [&dyn ToValue; 8] = [
+            &d.fixed,
+            &d.number,
+            &d.severity,
+            &d.description,
+            &0u32,
+            &icon_name,
+            &false,
+            &sensitive,
+        ];
+        store.set(&store.append(), &col_indices, &values);
+    }
+
+    store
+}
+
+fn fixed_toggled<W: IsA<gtk::CellRendererToggle>>(
+    model: &gtk::ListStore,
+    _w: &W,
+    path: gtk::TreePath,
+) {
+    let iter = model.get_iter(&path).unwrap();
+    let mut fixed = model
+        .get_value(&iter, Columns::Fixed as i32)
+        .get::<bool>()
+        .unwrap();
+    fixed = !fixed;
+    model.set_value(&iter, Columns::Fixed as u32, &fixed.to_value());
```

Comment:
Such a weird API: needs `u32` for the setter but `i32` for the getter :) Not your fault of course but...

See also https://blog.gtk.org/2019/02/08/report-from-the-gtk-hackfest-in-brussels/ the section about `GtkTreeView` btw. This is all hopefully going away completely with GTK4 and becomes a nicer API.